taipy/core/scenario/_scenario_manager.py

```diff
@@ -521,3 +521,70 @@ def _get_by_config_id(cls, config_id: str, version_number: Optional[str] = None)
         for fil in filters:
             fil.update({"config_id": config_id})
         return cls._repository._load_all(filters)
+
+    @classmethod
+    def _duplicate(cls, scenario: Scenario, creation_date: Optional[datetime] = None) -> Scenario:
```

Comment:
what about passing an optional name for the new scenario and an optional list of data nodes to duplicate? Do you plan to do it in another PR?

Reply:
yep I'm planning to do it in another PR to make it cleaner as the current one is already rather complicated

Reply:
ALl right. I let you create the potential issue if you need it.